addons/website/models/website_visitor.py

```diff
@@ -237,24 +268,25 @@ def _create_visitor(self, website_track_values=None):
             vals['website_track_ids'] = [(0, 0, website_track_values)]
         return self.sudo().create(vals)
 
-    def _cron_archive_visitors(self):
+    def _cron_archive_visitors_and_last_connections(self):
+        """ Since last_conection_datetime is in a temporary table (to avoid concurency update)
+            we need to clean them and update the visit count. """
         self.flush(['visit_count'])
-        yesterday = datetime.now() - timedelta(days=1)
         query = """
             UPDATE website_visitor
-            SET visit_count = visit_count + 1
-            WHERE id in (
-                SELECT visitor_id
-                FROM website_visitor_lastconnection
-                GROUP BY visitor_id
-                HAVING COUNT(*) > 1)
+            SET visit_count = (
+                SELECT COUNT(DISTINCT DATE(visit_datetime))
+                FROM website_track
+                WHERE date(visit_datetime) != DATE(%s)
```

Comment:
you can use CURRENT_DATE instead of DATE('datetime.now()')

Reply:
nice